core/services/gateway/handlers/vault/handler.go

```diff
@@ -207,7 +211,16 @@ func (h *handler) HandleJSONRPCUserMessage(ctx context.Context, req jsonrpc.Requ
 		return errors.New("request ID cannot be empty")
 	}
 
-	h.lggr.Debugw("handling vault request", "method", req.Method, "requestID", req.ID)
+	isAuthorized, owner, err := h.requestAuthorizer.AuthorizeRequest(ctx, req)
+	if !isAuthorized {
+		h.lggr.Errorw("request not authorized", "request_id", req.ID, "owner", owner, "reason:", err)
+		errors.New("request not authorized: " + err.Error())
+	}
+
+	// Prefix request id with owner, to ensure uniqueness across different owners
+	req.ID = owner + "::" + req.ID
```

Comment:
Are we validating that the owner or req.ID doesn't contain the separator?

Reply:
The owner is generated by our auth, and it will always be an ethereum address hex string which can not have this separator.
The req.ID can have this separator, and we wouldn't break, since while responding, we only strip this prefixed part which we added.